### PR TITLE
Rename component & extract store/reducer

### PR DESCRIPTION
**Pull request by arnaudgiuliani**
Just renaming components to stick more to unidirectional data flow concept

Extract datPublisher to hide it

### CODE REVIEW

File: uniflow-core/src/main/kotlin/io/uniflow/core/flow/UIDataStore.kt [hunk 1]
```diff
@@ -5,7 +5,7 @@ import io.uniflow.core.flow.data.UIState
 import io.uniflow.core.logger.UniFlowLogger
 
 
-class UIDataManager(private val publisher: UIDataPublisher, defaultState: UIState) {
+class UIDataStore(private val publisher: UIDataPublisher, defaultState: UIState) {
```
Comment (erikhuizinga) [inline uniflow-core/src/main/kotlin/io/uniflow/core/flow/UIDataStore.kt:8]: Good rename

File: uniflow-core/src/main/kotlin/io/uniflow/core/flow/UIDataPublisher.kt [hunk 2]
```diff
@@ -5,5 +5,5 @@ import io.uniflow.core.flow.data.UIState
 
 interface UIDataPublisher {
     suspend fun publishState(state: UIState)
-    suspend fun sendEvent(event: UIEvent)
+    suspend fun publishEvent(event: UIEvent)
```
Comment (erikhuizinga) [inline uniflow-core/src/main/kotlin/io/uniflow/core/flow/UIDataPublisher.kt:8]: Good rename

File: uniflow-test/src/test/kotlin/io/uniflow/test/ActorFlowTest.kt [hunk 3]
```diff
@@ -16,6 +16,7 @@ import io.uniflow.test.rule.TestDispatchersRule
 import io.uniflow.test.validate.validate
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
```
Comment (erikhuizinga) [inline uniflow-test/src/test/kotlin/io/uniflow/test/ActorFlowTest.kt:19]: Use `testCoroutineDispatcher.runBlockingTest` instead.

File: uniflow-test/src/test/kotlin/io/uniflow/test/StackFlowTest.kt [hunk 4]
```diff
@@ -12,6 +12,7 @@ import io.uniflow.test.impl.SampleFlow
 import io.uniflow.test.rule.TestDispatchersRule
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
```
Comment (erikhuizinga) [inline uniflow-test/src/test/kotlin/io/uniflow/test/StackFlowTest.kt:15]: Use `testCoroutineDispatcher.runBlockingTest` instead.

File: uniflow-core/src/main/kotlin/io/uniflow/core/flow/ActionReducer.kt [hunk 5]
```diff
@@ -14,8 +14,8 @@ import kotlinx.coroutines.isActive
 import kotlinx.coroutines.withContext
 
 @UseExperimental(ObsoleteCoroutinesApi::class)
-class ActionFlowScheduler(private val uiDataManager: UIDataManager, private val coroutineScope: CoroutineScope,
-    private val defaultDispatcher: CoroutineDispatcher, defaultCapacity: Int = Channel.BUFFERED) {
+class ActionReducer(private val uiDataManager: UIDataStore, private val coroutineScope: CoroutineScope,
```
Comment (erikhuizinga) [inline uniflow-core/src/main/kotlin/io/uniflow/core/flow/ActionReducer.kt:17]: You could rename `uiDataManager` too:

```suggestion
class ActionReducer(private val uiDataStore: UIDataStore, private val coroutineScope: CoroutineScope,
```

**Comment by arnaudgiuliani** [comment]
Finally got a way to write an Action Dispatcher and don't allow override of action functions

**Pull request merged**